book/internals/wrap_gtk_widget.nim

```diff
@@ -0,0 +1,192 @@
+
+For a short example of that, look at owlkettle's [counter-example](https://github.com/can-lehmann/owlkettle/blob/main/examples/counter.nim).
+
+
+### *Add features to the Widget*
+Beyond just creating the widget and providing event-listeners, gtk may provide options to further customize a widget.
+
+E.g. you may be able to show or hide values, change their positioning, enable or disable them etc. by updating certain fields at runtime after the widget was constructed.
+
+Take a look at the procs of your widget (and the procs of its parent classes) in the gtk docs, specifically anything that isn't a getter.
+If there's procs to add or set a property of the widget, try to add that feature to your wrapped widget.
+
+At minimum, try to wrap all features exposed by your widget directly and possibly its direct parent (e.g. for `Gtk.Scale` everything from there as well as its parent `Gtk.Range`)
+
+Follow and repeat the following steps to add a feature:
+- Wrap the gtk procs as explained in earlier sections
+- Add a field to your widget for every parameter required by the wrapped gtk procs.
+  If a field already exists because you use it during initialization with the constructor, then you don't need to add a new field.
+- Add a `property` hook for every field on your widget that you have a wrapped gtk proc for.
+  The hook should do nothing but call the wrapped gtk proc with values from the implicitly available `state` variable.
+- Run your example application to test whether the added feature does or does not work.
+
+### *Add and update documentation*
+Once you are satisfied with your widget, it's time to add a bit more documentation.
+
+##### 1) Add documentation and examples
+Add doc comments to properties and event callbacks to explain what they do.
+
+Further you can add examples using owlkettle's `example` macro to demonstrate usage of the widget.
+These will get automatically added to widget.ms in the next step
+
+For examples that use the `example` macro, search for `, example:` in [widgets.nim](https://github.com/can-lehmann/owlkettle/blob/main/owlkettle/widgets.nim).
+
+
+##### 2) Update widgets.md and examples/README.md
+Once that is done, run `nimble genDocs` to add your new widgets, doc comments, and examples to widgets.md, the main documentation file for widgets. 
+Once finished, commit the updated widgets.md.
+
+Next take a screenshot of your example application against a white background and put it in `docs/assets/examples`.
+Lastly, add an entry for your new widget to the widgets table of `examples/README.md` like so (Remember to replace the `<Widgetname>`, `<WidgetExample>` and `<WidgetExampleImage>` placeholders):
+```html
+    <tr>
+    <td>
+      <a href="https://github.com/can-lehmann/owlkettle/blob/main/examples/widgets/<WidgetExample>.nim">
+        <WidgetName>
+      </a>
+    </td>
+    <td><img alt="<WidgetName>" src="../docs/assets/examples/<WidgetExampleImage>.png" width="757px"></td>
```

Comment:
Please use the same code formatting as the existing entries. Mention that the width needs to half of the screenshot width.